driverdataaggregation/main.py

```diff
@@ -0,0 +1,601 @@
+                "WRITE_TO_DIR"
+            ] = aggregation_consumer_sessions.run_settings.env_vars[
+                "READ_FROM_DIR"
+            ] = shared_scratch_dir
+
+            exp.start(aggregation_producer_sessions,
+                      aggregation_consumer_sessions,
+                      summary=True)
+
+            # confirm scaling test run successfully
+            stat = exp.get_status(aggregation_producer_sessions)
+            if stat[0] != status.STATUS_COMPLETED:
+                logger.error(f"ERROR: One of the scaling tests failed \
+                                {aggregation_producer_sessions.name}")
+            stat = exp.get_status(aggregation_consumer_sessions)
+            if stat[0] != status.STATUS_COMPLETED:
+                logger.error(f"ERROR: One of the scaling tests failed \
+                                {aggregation_consumer_sessions.name}")
+
+    @classmethod
+    def create_aggregation_producer_session_python_fs(cls, exp, nodes, tasks, iterations,
+                                                  bytes_, t_per_dataset):
+        return cls._create_aggregation_producer_session(
+            name="aggregate-sess-prod-for-python-consumer-file-system",
+            exe="./cpp-py-data-aggregation/fs/build/aggregation_producer",
+            exe_args=[str(bytes_), str(t_per_dataset)],
+            exp=exp, nodes=nodes, tasks=tasks, db_nodes=0, db_cpus=0,
+            iterations=iterations, bytes_=bytes_, t_per_dataset=t_per_dataset)
+    
+    @classmethod
+    def create_aggregation_consumer_session_python_fs(cls, exp, nodes, tasks, iterations,
+                                                  bytes_, t_per_dataset, c_threads,
+                                                  cpu_hyperthreads):
+        py_script_dir = "./cpp-py-data-aggregation/fs/"
+        py_script = "aggregation_consumer.py"
+        model = cls._create_aggregation_consumer_session(
+            name="aggregate-sess-cons-python",
+            exe=sys.executable,
+            exe_args=[py_script, str(nodes*tasks)],
+            exp=exp, nodes=nodes, tasks=tasks, db_nodes=0, db_cpus=0,
+            iterations=iterations, bytes_=bytes_, t_per_dataset=t_per_dataset,
+            c_threads=c_threads, cpu_hyperthreads=cpu_hyperthreads)
+        model.attach_generator_files(to_copy=[py_script_dir + py_script])
+        exp.generate(model, overwrite=True)
+        write_run_config(model.path,
+                        client_total=tasks*nodes,
+                        client_per_node=tasks,
+                        client_nodes=nodes,
+                        database_nodes=0,
+                        database_cpus=0,
+                        iterations=iterations,
+                        tensor_bytes=bytes_,
+                        t_per_dataset=t_per_dataset,
+                        client_threads=c_threads)
+        return model
+
+if __name__ == "__main__":
+    import sys
+    sys.path.append('..')
```

Comment:
I was planning on incorporating it into this PR! When you say fold it into the next if stmt w/ the same conditional, you mean combine the if statements since they both have __main__ -> it wont conflict?? Both are messing with sys right? I thought that by having them separate, it would run one if statement if in home dir but other if cd

Reply:
Oh! I just mean change:
```py
if __name__ == "__main__":
    import sys
    sys.path.append("..")

if __name__ == "__main__":   # <-- redundant extra check we already made above
    import fire
    fire.Fire(DataAggregation())
```
into
```py

if __name__ == "__main__":
    import sys
    import fire
    sys.path.append("..")
    fire.Fire(DataAggregation())
```

Reply:
unrelated side note, if we are going to allow for `cd driverdataaggregation; python main.py aggregation_scaling` the `sys.path` manipulation will need to take place before we try and import something from the parent dir (i.e. before we try to `import utils`)

Reply:
discussed